src/hocon.erl

```diff
@@ -306,6 +306,27 @@ lookup(Var, [_Other|More], ResolvedValue) ->
 lookup(_Var, [], ResolvedValue) ->
     ResolvedValue.
 
+typeof({concat, [{concat, X}|_More]}) ->
+    typeof({concat, X});
+typeof({concat, [X|_]}) when is_list(X) ->
+    array;
+typeof({concat, [{X}|_]}) when is_list(X) ->
+    object;
+typeof({concat, _Other}) ->
+    string.
+
+maybe_merge({concat, Old}, {concat, New}) ->
+    case {typeof({concat, Old}), typeof({concat, New})} of
+        {array, array} ->
```

Comment:
since we are trying to look up a field value here,
the field value can only be concatenated when it's an object, (or object of objects),
but can never be concatenated when it's an array, (or array of arrays).

```
is_object([{concat, X}|_]) -> is_object(X);
is_object([{X}|_]) -> is_list(X);
is_object(_) -> false.
```
Then only append `Old` and `New` when they are both objects.